include/zephyr/sys/util.h

```diff
@@ -110,6 +110,44 @@ extern "C" {
 
 #endif /* __cplusplus */
 
+/**
+ * @brief Whether @p ptr is an element of @p array
+ *
+ * This macro can be seen as a slightly stricter version of @ref PART_OF_ARRAY
+ * in that it also ensures that @p ptr is aligned to an array-element boundary
+ * of @p array.
+ *
+ * In C, passing a pointer as @p array causes a compile error.
+ *
+ * @param array the array in question
+ * @param ptr the pointer to check
+ *
+ * @return 1 if @p ptr is part of @p array, 0 otherwise
+ */
+#define IS_ARRAY_ELEMENT(array, ptr)                                                               \
+	((ptr) && POINTER_TO_UINT(&(array)[0]) <= POINTER_TO_UINT(ptr) &&                          \
```

Comment:
Any reason for the `&(array)[0]`? Wouldn't that always be the same as `(array)`? Just for semantic reasons?

Reply:
Sure - I can change that. It was copied from the original implementation of `PART_OF_ARRAY`